storage-api/src/main/java/com/linagora/calendar/storage/configuration/resolver/SafeUserSettingsResolver.java

```diff
@@ -0,0 +1,73 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.calendar.storage.configuration.resolver;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.MailboxSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linagora.calendar.storage.SimpleSessionProvider;
+import com.linagora.calendar.storage.exception.DomainNotFoundException;
+
+import reactor.core.publisher.Mono;
+
+public class SafeUserSettingsResolver {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SafeUserSettingsResolver.class);
+
+    private final SimpleSessionProvider sessionProvider;
+    private final SettingsBasedResolver settingsBasedResolver;
+
+    public SafeUserSettingsResolver(SimpleSessionProvider sessionProvider,
+                                    SettingsBasedResolver settingsBasedResolver) {
+        this.sessionProvider = sessionProvider;
+        this.settingsBasedResolver = settingsBasedResolver;
+    }
+
+    public Mono<SettingsBasedResolver.ResolvedSettings> get(Username user) {
+        return get(sessionProvider.createSession(user));
+    }
+
+    public Mono<SettingsBasedResolver.ResolvedSettings> get(MailboxSession session) {
+        return settingsBasedResolver.readSavedSettings(session)
+            .onErrorResume(error -> {
+                logError(session.getUser(), error);
+                return Mono.just(SettingsBasedResolver.ResolvedSettings.DEFAULT);
+            })
+            .switchIfEmpty(Mono.just(SettingsBasedResolver.ResolvedSettings.DEFAULT));
+    }
+
+    public Mono<SettingsBasedResolver.ResolvedSettings> getWithFallback(Username user, Username secondUser) {
```

Comment:
IMO again this can be turned into a top level method within `SettingsBasedResolver`